src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -194,11 +189,14 @@ private void initialize(Context appContext, String authority, ITokenCacheStore t
             throw new UnsupportedOperationException("Local cache is not supported for broker usage");
         }
         mContext = appContext;
-        mConnectionService = new DefaultConnectionService(mContext);
         checkInternetPermission();
         mAuthority = extractAuthority(authority);
         mValidateAuthority = validateAuthority;
         mTokenCacheStore = tokenCacheStore;
+        
+        if (mTokenCacheStore != null) {
```

Comment:
do not check this for null... create token accessor anyway, and within token accessor check if token store is null
And don't check if mTokenCacheAccessor is null anywhere